payments-core/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt

```diff
@@ -56,7 +56,7 @@ internal class DefaultFlowController internal constructor(
     private val paymentOptionFactory: PaymentOptionFactory,
     private val flowControllerInitializer: FlowControllerInitializer,
     paymentControllerFactory: PaymentControllerFactory,
-    paymentFlowResultProcessorFactory: (String, StripeApiRepository) -> PaymentFlowResultProcessor,
+    paymentFlowResultProcessorFactory: (String, StripeApiRepository) -> PaymentFlowResultProcessor<PaymentIntentResult>,
```

Comment:
will we add another factory for `SetupIntentResult` later? if so probably worthwhile to rename this to something like `paymentIntentFlowResultProcessorFactory`

Reply:
Also should it be type `PaymentIntentFlowResultProcessor` ?